scripts/mixins/families/aern.lua

```diff
@@ -20,21 +20,51 @@ g_mixins.families.aern = function(aernMob)
                     reraises = 1
                 end
             end
+            mob:setMobMod(xi.mobMod.NO_DROPS, 0) -- Drops on if Not reraising
             if curr_reraise < reraises then
+                mob:setMobMod(xi.mobMod.NO_DROPS, 1)  -- Drops off if reraising
                 local dropid = mob:getDropID()
-                mob:setDropID(0)
+                local owner = nil
```

Comment:
`local thing` is valid syntax for defining a local var as nil, but we don't even need it here as `target` is already a local var on the next line and not a function arg like killer was we can just redefine it as needed and use it instead of owner. `target = target:getMaster()` legit works and then the rest of the code can safely assume its not a pet and not nil.

Reply:
So I did this because it catches a specific case:

We don't automatically want to set the target as the owner of the pet in the case that the pet is still alive and well, Aerns should if possibly reaggro whoever the target was. 

I set up the owner nil scenario as a placeholder for the owner in case the target was dead and was a pet, and then and only then should we aggro the owner.

Maybe i'm mis-reading what you are saying.  

Are you saying on the owner ~= nil line we can:

target:getMaster() ~= nil 

instead? (I would get that if the target isn't a pet getMaster() returns nil?)

Reply:
> We don't automatically want to set the target as the owner of the pet in the case that the pet is still alive and well, Aerns should if possibly reaggro whoever the target was.

that's not what we are doing though. we don't need to know the owner beyond ensuring we switch target TO the owner if target was a pet. this happens straight away so that all code after is dealing with a player only and never a pet.
```lua
local target = mob:getTarget()
if not target then
     -- hold up target doesn't exist
end
if target:isPet() then
    -- Whoa targets a pet! we don't wnat that!
    target = target:getMaster()
end
if target:isDead() then
    -- Whoa targets dead! we need a new one
end
```
etc

Reply:
the whole: 
```lua
if target:isPet() then
    -- Whoa targets a pet! we don't wnat that!
    target = target:getMaster()
end
```

We do want a pet though, if the pet is alive and well, the aern SHOULD aggro the pet and not the master.

Reply:
Which is why I left the original stuff, the current script just cleans it up a bit.

If i'm a pet and not alive, then target should be my master.

If my master isn't alive then ill skip the first if and goto the 'find a valid target in my alliance' loop.

If all else fails, deaggro.

Reply:
I was sure it went for the master on retail. that changes things and is a pain because they share a claim and functions that don't work on pets etc.

will come back to this in a bit have to afk

Reply:
confirmed in game by WinterSolstice8 that Aerec is correct - its just difficult for the pet to have more hate than the master, but it goes for highest hate.